client-spec/client-specification-v1.md

```diff
@@ -0,0 +1,322 @@
+as an atomic operation
+(i.e, it either succeeds or fails completely).
+
+
+## Module relationships
+
+Modules can be put in relationship with one another using one or more relationship descriptors.
+
+There are 4 types of relationship fields:
+1. `depends`
+2. `recommends`
+3. `suggests`
+4. `conflicts`
+
+These 4 fields are all optional.
+If present, they contain a list of relationship descriptors with the following fields:
+- `name`: contains the `identifier` of the related module (required);
+- `version`: the specific version of the related module (optional);
+- `min_version` and `max_version`: allows to specify a range of versions (optional).
+
+Version ranges are only valid for the `conflicts` relationship:
+if version ranges are encountered in other relationship descriptors,
+the behaviour is undefined: clients may choose how to resolve the ambiguity.
+
+### depends
+
+Indicates that a module cannot be installed unless all its dependencies are not installed too.
+Clients must not install a module unless all of its dependencies are either installed,
+or are about to be installed alongside the module.
+
+Clients may implement a way for the user to override the dependency checking:
+this mechanism, if implemented, must be explicitly selected by the user
+for each install process.
+
+### recommends
+
+A client must automatically select for installation all the packages
+that are recommended by any of the modules about to be installed.
+The selection must not be recursive:
+packages that are recommended by a package that was recommended
+must not be selected too.
+
+Clients should implement a way for the user to refuse one or more recommended packages,
+and may implement a way to enable recursive recommendations.
+
+### suggests
+
+A client must not automatically select for installation all the packages
+that are suggested by any of the modules about to be installed.
+The selection must not be recursive:
+packages that are recommended by a package that was suggested
+must not be selected too.
+
+Clients should implement a way for the user to select one or more suggested packages,
+and may implement a way to enable recursive suggestions.
+
+### conflicts
+
+Indicates that two or more modules cannot be installed at the same time.
+Clients must not allow installation of conflicting modules at the same time.
+
+Clients may implement a way for the user to override the conflict checking:
+this mechanism, if implemented, must be explicitly selected by the user
+for each install process.
+
+### provides
+
+Additionally, modules may indicate they provide a name.
+Providing must be used to signal interchangeable modules:
+modules may be in relationships with a provided name.
+
+Version constraints are not valid when referring to a provided name.
+
+If a package conflicts with a provided name,
+clients must consider all the packages that provide that name
+as conflicting and deal with them as described in the `conflicts` section.
```

Comment:
Actually, it does: for example FAR [conflicts with AeroDynamicModel](https://github.com/KSP-CKAN/CKAN-meta/blob/master/FerramAerospaceResearch/FerramAerospaceResearch-v0.15.2_Ferri.ckan#L15-L18), which makes it incompatible with any other mod that provides that name (for example, NEAR).